src/model/validation/message.ts

```diff
@@ -76,3 +190,42 @@ function severityToString(severity: Severity) {
             return 'Compatibility issue';
     }
 }
+
+function calculateIsSuppressed(
+    severity: Severity,
+    location: AstNodeWithDirectives | undefined,
+    code: MessageCode,
+) {
+    const suppressDirective = location?.directives?.find(
+        (d) => d.name.value === SUPPRESS_DIRECTIVE,
+    );
+    if (!suppressDirective) {
+        return false;
+    }
+    let argName;
+    switch (severity) {
+        case Severity.COMPATIBILITY_ISSUE:
+            argName = SUPPRESS_COMPATIBILITY_ISSUES_ARG;
+            break;
+        case Severity.WARNING:
+            argName = SUPPRESS_WARNINGS_ARG;
+            break;
+        case Severity.INFO:
+            argName = SUPPRESS_INFOS_ARG;
+            break;
+        default:
+            throw new Error(`Non-suppressable severity: ${severity}`);
```

Comment:
Where is this catched? Shouldn't the method simply return `false` in this case?

Reply:
It's a code path that should never occur (it would mean that some code reported a suppressable error)